File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int curSum = nums[0];
        int maxSoFar = nums[0];
        int n = nums.size();
        for(int i=1;i<n;i++){
            if(curSum<0) curSum=0;
            curSum+=nums[i];
            
            maxSoFar = max(maxSoFar,curSum);
        }
        
        return maxSoFar;
    }
};

File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curSum,maxSum = nums[0],nums[0]
        for elem in nums[1:]:
            if curSum<0:
                curSum=0
            curSum+=elem
            maxSum = max(maxSum,curSum)
        return maxSum